rust/agama-lib/src/software/client.rs

```diff
@@ -1,71 +1,78 @@
-use super::proxies::SoftwareProductProxy;
+use super::proxies::Software1Proxy;
 use crate::error::ServiceError;
 use serde::Serialize;
 use zbus::Connection;
 
 /// Represents a software product
 #[derive(Debug, Serialize)]
-pub struct Product {
-    /// Product ID (eg., "ALP", "Tumbleweed", etc.)
+pub struct Pattern {
+    /// Pattern ID (eg., "aaa_base", "gnome")
     pub id: String,
-    /// Product name (e.g., "openSUSE Tumbleweed")
-    pub name: String,
-    /// Product description
+    /// Pattern category (e.g., "Production")
+    pub category: String,
+    /// Pattern icon path locally on system
+    pub icon: String,
+    /// Pattern description
     pub description: String,
+    /// Pattern summary
+    pub summary: String,
+    /// Pattern order
+    pub order: String,
 }
 
 /// D-Bus client for the software service
 pub struct SoftwareClient<'a> {
-    product_proxy: SoftwareProductProxy<'a>,
+    software_proxy: Software1Proxy<'a>,
 }
 
 impl<'a> SoftwareClient<'a> {
     pub async fn new(connection: Connection) -> Result<SoftwareClient<'a>, ServiceError> {
         Ok(Self {
-            product_proxy: SoftwareProductProxy::new(&connection).await?,
+            software_proxy: Software1Proxy::new(&connection).await?,
         })
     }
 
-    /// Returns the available products
-    pub async fn products(&self) -> Result<Vec<Product>, ServiceError> {
-        let products: Vec<Product> = self
-            .product_proxy
-            .available_products()
+    /// Returns the available patterns
+    pub async fn patterns(&self, filtered: bool) -> Result<Vec<Pattern>, ServiceError> {
```

Comment:
`filtered` is not documented here nor in https://github.com/openSUSE/agama/blob/39534694aed2b3ce941e4c4b896051ab1cd0660b/doc/dbus/org.opensuse.Agama.Software1.doc.xml#L8

Filtered according to what?

Not your fault ;-) but let's fix it now

Reply:
done in https://github.com/openSUSE/agama/pull/919/commits/b75a6e14a2fd1108aa6cf8bf0bba8c2fef8e25dc